mono/utils/mono-static-mutex.h

```diff
@@ -0,0 +1,80 @@
+/* -*- Mode: C; tab-width: 8; indent-tabs-mode: t; c-basic-offset: 8 -*- */
+/**
+ * \file
+ * Initializers for PRIO_INHERIT mutexes
+ * Based on mono-os-mutex.h
+ *
+ * Authors: Jeffrey Stedfast <fejj@ximian.com>
+ *          Kurt Van Dijck <dev.kurt@vandijck-laurijssen.be>
+ *
+ * Copyright 2002 Ximian, Inc. (www.ximian.com)
+ *
+ * Licensed under the MIT license. See LICENSE file in the project root for full license information.
+ */
+
+#ifndef __MONO_STATIC_MUTEX_H__
+#define __MONO_STATIC_MUTEX_H__
+
+#if !defined(HOST_WIN32)
+#include <pthread.h>
+#include <errno.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#define st_error(FMT, ...) \
```

Comment:
`mono-os-mutex.h` uses `g_error` here.  Is there some reason you can't use `g_error` or `g_assertf`?

Reply:
This was my approach to let eglib not depend on mono code.